execution.bs

```diff
@@ -4275,11 +4342,11 @@ enum class forward_progress_guarantee {
     <pre highlight=c++>
     template&lt;class T>
       concept receiver =
-        move_constructible&lt;remove_cvref_t&lt;T>> &&
-        constructible_from&lt;remove_cvref_t&lt;T>, T> &&
         requires(const remove_cvref_t&lt;T>& t) {
-          execution::get_env(t);
-        };
+          { execution::get_env(t) } -> queryable;
+        } &&
+        move_constructible&lt;remove_cvref_t&lt;T>> &&
+        constructible_from&lt;remove_cvref_t&lt;T>, T>;
```

Comment:
TODO: add an editorial note here about whether the constructibility constraints should instead be mandates.